lib/gst/player/gstplayer.c

```diff
@@ -1121,6 +1144,458 @@ element_cb (GstBus * bus, GstMessage * msg, gpointer user_data)
   }
 }
 
+static void
+player_set_flag (GstPlayer *self, gint pos)
+{
+  gint flags;
+
+  g_object_get (self->priv->playbin, "flags", &flags, NULL);
+  flags |= pos;
+  g_object_set (self->priv->playbin, "flags", &flags, NULL);
+}
+
+static void
+player_clear_flag (GstPlayer *self, gint pos)
+{
+  gint flags;
+
+  g_object_get (self->priv->playbin, "flags", &flags, NULL);
+  flags &= ~pos;
+  g_object_set (self->priv->playbin, "flags", &flags, NULL);
+}
+
+static gboolean
+is_track_enabled (GstPlayer *self, gint pos)
+{
+  gint flags;
+
+  g_object_get (self->priv->playbin, "flags", &flags, NULL);
+  if (flags & pos)
+    return TRUE;
+
+  return FALSE;
+}
+
+typedef struct
+{
+  GstPlayer *player;
+  GstPlayerMediaInfo  *info;
+  guint signal;
+}MediaInfoUpdatedSignalData;
+
+static gboolean
+media_info_updated_dispatch (gpointer user_data)
+{
+  MediaInfoUpdatedSignalData  *data = user_data;
+
+  g_signal_emit (data->player, signals[data->signal], 0, data->info);
+
+  return FALSE;
+}
+
+static void
+free_media_info_updated_signal_data (MediaInfoUpdatedSignalData *data)
+{
+  g_slice_free (MediaInfoUpdatedSignalData, data);
+}
+
+static void
+emit_media_updated_signal (GstPlayer *self, GstPlayerMediaInfo *info,
+  guint signal)
+{
+  if (self->priv->dispatch_to_main_context) {
+    MediaInfoUpdatedSignalData *data = g_slice_new
+                                        (MediaInfoUpdatedSignalData);
+
+    /* unref the previous media info object */
+    if (self->priv->prev_media_info)
+      g_object_unref (self->priv->prev_media_info);
+    self->priv->prev_media_info = self->priv->media_info;
+    self->priv->media_info = info;
+
+    data->player = self;
+    data->info   = self->priv->media_info;
+    data->signal = signal;
+
+    g_main_context_invoke_full (self->priv->application_context,
+                    G_PRIORITY_DEFAULT,
+                    media_info_updated_dispatch,
+                    data,
+                    (GDestroyNotify) free_media_info_updated_signal_data);
+  } else {
+    g_signal_emit (self, signals[signal], 0);
```

Comment:
In this case you're emitting the signal without the actual media info, and also you don't update self->priv->media_info